2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -40,47 +40,45 @@ function cache_container_addr() {
 
 # wait_for_mongo_up waits until the mongo server accepts incomming connections
 function wait_for_mongo_up() {
-  local mongo_host
-  mongo_host="${1-}"
-  local mongo_cmd
-  mongo_cmd="mongo admin "
-
-  if [ ! -z "${mongo_host}" ]; then
-    mongo_cmd+="--host ${mongo_host}:${CONTAINER_PORT} "
-  fi
-
-  for i in $(seq $MAX_ATTEMPTS); do
-    echo "=> Waiting for MongoDB service startup ${mongo_host} ..."
-    set +e
-    $mongo_cmd --eval "help" &>/dev/null
-    status=$?
-    set -e
-    if [ $status -eq 0 ]; then
-      echo "=> MongoDB service has started"
-      return 0
-    fi
-    sleep $SLEEP_TIME
-  done
-  echo "=> Giving up: Failed to start MongoDB service!"
-  exit 1
+  wait_mongo 1
 }
 
 # wait_for_mongo_down waits until the mongo server is down
 function wait_for_mongo_down() {
+  wait_mongo 0
+}
+
+# wait_mongo waits until the mongo server is up/down
+# $1 - 0 or 1 - to specify for what to wait (0 - down, 1 - up)
+# $2 - host where to connect (localhost by default)
+function wait_mongo() {
+  local operation
+  operation=-eq
+  local message
+  message="up"
```

Comment:
According to https://google.github.io/styleguide/shell.xml?showone=Use_Local_Variables#Use_Local_Variables the reason is that local discards the return value of a command executed during the expansion. This does not happen here, so it's OK to use just one line and in my opinion it will look a much better.